File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
import java.util.ArrayList;
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        ArrayList<String> board1 = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            String tmp = String.format("%0" + n + "d", Long.parseLong(Integer.toBinaryString(arr1[i]).toString()));
            board1.add(tmp);
        }
        
        ArrayList<String> board2 = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            String tmp = String.format("%0" + n + "d", Long.parseLong(Integer.toBinaryString(arr2[i]).toString()));
            board2.add(tmp);
        }
        
        for (int i = 0; i < n; i++) {
            String tmp = "";
            for (int j = 0; j < n; j++) {
                if (board1.get(i).charAt(j) == '0' && board2.get(i).charAt(j) == '0') {
                    tmp += " ";
                } else {
                    tmp += "#";
                }
            }
            answer[i] = tmp;
        }
        return answer;
    }
}

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    new_board = [['#'] * n for _ in range(n)]
    
    board1 = []
    for i in range(n):
        num = arr1[i]
        a = format(num, "b")
        board1.append(a.zfill(n))
    print(board1)
        
    board2 = []
    for i in range(n):
        num = arr2[i]
        a = format(num, "b")
        board2.append(a.zfill(n))
        
    for i in range(n):
        for j in range(n):
            if board1[i][j] == '0' and board2[i][j] == '0':
                new_board[i][j] = ' '
                
    for i in range(n):
        answer.append(''.join(new_board[i]))
    

    return answer